### ISSUE TITLE
Template engine of content affects front matter of layout

### ISSUE THREAD

**Issue by thejohnfreeman** [11ty/eleventy#446]
The inputs:

- In my template: `templateEngineOverride: md`
- In the template's directory data file: `layout: post.njk`
- In `post.njk`: `permalink: "blog/{{ page.date | date: '%Y/%m/%d' }}/{{ title | slug }}/index.html"`

The output:

```
Writing _site/<p>blog/{{ page.date | date: '%Y/%m/%d' }}/{{ title | slug }}/index.html</p>
 from ./src/blog/2012-03-23-template-inheritance-for-handlebars.md.
```

Totally unexpected. ~~No actual file is written at that location, of course. Perhaps Eleventy swallows the "not a legal file name" error?~~ (I was looking in the wrong directory.)

If I set `markdownTemplateEngine` in the `.eleventy.js` configuration, the same thing happens, which means every post overwrites the same file.

**Comment by thejohnfreeman** [comment]
For users: my workaround to disable the template language for my content is to wrap it all, head to toe, in `{% raw %} ... {% endraw %}`.

**Comment by zachleat** [comment]
Hmm… I agree that is is unfortunate. It does seem strange to me that you’d use a nunjucks file but not want nunjucks rendered?

It seems almost as though you want an engine override to apply only to the permalink. Something like `permalinkTemplateEngineOverride`.

It will take some persuasion to classify this as a bug and not a feature request though.

**Comment by zachleat** [comment]
Oh oh, hmm I misread your issue and glossed over your use of `layout` here entirely, sorry about that.

**Comment by zachleat** [comment]
Created a repro gist https://gist.github.com/zachleat/f199e9b0856337506e13a486d96514ad that exhibited this issue with Eleventy 0.7.1. This issue was fixed with Eleventy 0.8.0.

Sorry for the late reply! 0.8.0 was released April 2019 https://github.com/11ty/eleventy/releases/tag/v0.8.0
